sources/apigateway/resource.go

```diff
@@ -0,0 +1,97 @@
+package apigateway
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/apigateway"
+	"github.com/aws/aws-sdk-go-v2/service/apigateway/types"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+func convertGetResourceOutputToResource(output *apigateway.GetResourceOutput) *types.Resource {
+	return &types.Resource{
+		Id:              output.Id,
+		ParentId:        output.ParentId,
+		Path:            output.Path,
+		PathPart:        output.PathPart,
+		ResourceMethods: output.ResourceMethods,
+	}
+}
+
+func resourceOutputMapper(scope string, awsItem *types.Resource) (*sdp.Item, error) {
+	attributes, err := sources.ToAttributesCase(awsItem, "tags")
+	if err != nil {
+		return nil, err
+	}
+
+	item := sdp.Item{
+		Type:            "apigateway-resource",
+		UniqueAttribute: "id",
```

Comment:
We need to create a new combined ID here if we're going to use `rest-api-id/resource-id` as the input for GET. A GET request should always result in an item with *exactly* the same UniqueAttributeValue as what the user searched for

Reply:
This was something that I wanted to ask.
So the get query must include the `rest-api-id` but the [`*types.Resource`](https://github.com/aws/aws-sdk-go-v2/blob/main/service/apigateway/types/types.go#L966)that we use to map AWS object to sdp item does not have the `rest-api-id`.

What should we do in that case?

Reply:
https://github.com/overmindtech/aws-source/pull/608/commits/9381c53353962b3ef0c4470af5e3eab87a74f9cd